torch/csrc/inductor/aoti_torch/shim_common.cpp

```diff
@@ -55,6 +56,14 @@ c10::optional<T> pointer_to_optional(U* ptr) {
 
 } // namespace
 
+int64_t aoti_torch_div_floor_int64(int64_t a, int64_t b) {
```

Comment:
it appears we only use `div_floor_integer<int64_t>` when emitting indexing expressions, so I only created this particular instantiation (since C doesn't support overloading)

Reply:
I think I was doing the same thing for `div_floor_integer` at some point, but realized we need a way to capture exception here, like `divide-by-zero`. So making the API similar to others here by using `AOTI_TORCH_CONVERT_EXCEPTION_TO_ERROR_CODE` seems inevitable, unless @ezyang has a better idea.

Reply:
@swolchok if you're doing a CPU model and there's some code where we do an integer divide, which potentially may have a zero denominator. What would you prefer the AOTInductor DSO behavior to be in this case?

Reply:
I don't understand -- div_floor_integer doesn't check for divide by zero.

I also want to clear up what is supposed to happen -- integer division by zero has undefined behavior (https://en.cppreference.com/w/c/language/operator_arithmetic#Division). it is not guaranteed to crash the process. it is very unlikely to raise a C++ exception. C++ authors must guarantee that their code does not perform integer division by zero.

Reply:
supposing that div_floor_integer *did* check for divide by zero, I would prefer that it was compiled into the DSO, not going through an aoti_torch shim, because those are relatively expensive and it is a small function.

Reply:
in particular we could simply emit a zero check for the denominator inline and throw an exception, which would be translated into an error code at the boundary and then back into an exception on the other side

Reply:
I guess regardless of whether it does check or not, it's better to have it compiled into the DSO and/or inlined? Let me look into that...

Reply:
> compiled into the DSO and/or inlined?

can't inline across a DSO boundary, gotta compile into the DSO

Reply:
Yeah I didn't really phrase things correctly... should not have said "or". I just meant that if we compile the `div_floor_*` functions into the DSO (I assume that means model.so), then they could possibly be inlined into their callers.